### ISSUE TITLE
Retry middleware should only retry idempotent HTTP methods (not POST and PATCH)

### ISSUE THREAD

**Issue by hoh** [traefik/traefik#8990]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

The retry middleware should by default only retry idempotent HTTP methods (`OPTION`, `GET`, `HEAD`, `PUT`, and `DELETE`). Retrying `POST` and `PATCH` can lead to issues on the backend.

## Context
While _"As soon as the server answers, the middleware stops retrying"_, in HTTP the server does not have to start answering before it starts to process the request. This means that, for non idempotent methods, the server may change state in a way that should not be called multiple times, and this before starting to answer.

This can lead to issues such as double-spending, multiple object creation, ... and should be avoided by default.

An option to allow the middleware to retry on non idempotent methods would be a nice to have.

Duplicate of the frozen-due-to-age: https://github.com/traefik/traefik/issues/6942

**Comment by lplazas** [comment]
@kevinpollet is this something you will be working on? If not I'd love to have a go at to get kickstarted in the codebase.

**Comment by github-sys-admin** [comment]
go ahead

**Comment by kevinpollet** [comment]
Hello @lplazas,

Thanks for asking, we discussed it with the team and someone is already working on the subject. Feel free to take another subject, help is always welcome.

**Comment by lplazas** [comment]
Thank you for your reply @kevinpollet. Seems like last Friday I was trolled by someone who posted that I could go ahead with this task, but now that comment is deleted. 
I worked on this over the weekend so I'll anyways open a PR, feel free to check it if you think it's worth it or use it as a base for the final change: https://github.com/traefik/traefik/pull/8998

**Comment by kevinpollet** [comment]
Hello @hoh,

Thanks for your interest in Traefik,

As stated in https://github.com/traefik/traefik/pull/8998#issuecomment-1123705126, the middleware will not retry a request if one byte of data is sent to the backend. We had wrongly classified this issue as a legitimate feature.

According to this, we will decline this feature request.
